assets/auth/firebase/reset_password.ts

```diff
@@ -0,0 +1,43 @@
+import {auth} from './init';
+import {sendPasswordResetEmail} from 'firebase/auth';
+
+declare const nextUrl: string;
+
+const form = document.getElementById('formToken') as HTMLFormElement;
+const url = new URL(nextUrl);
+const params = new URLSearchParams(url.search);
+const sendButton = document.getElementById('reset-password-btn') as HTMLButtonElement;
+
+if (sendButton != null) {
+    form.onsubmit = (event) => {
+        event.preventDefault();
+
+        if (sendButton.disabled) {
+            return false;
+        }
+
+        const data = new FormData(form);
+        const email = data.get('email') as string;
+
+        params.append('email', email);
+        url.search = params.toString();
+
+        sendButton.disabled = true;
+        sendPasswordResetEmail(auth, email, {url: url.toString()})
+            .then(() => {
+                // set `email_sent` to true, so server knows password reset was handled by firebase
```

Comment:
I see `email_sent` is no longer used, as we're checking the auth provider on the server side. Therefor this comment is no longer required